### ISSUE TITLE
no warning when `dataclass` extends non-dataclass with an `__init__` method

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#13051]
```py
from dataclasses import dataclass


class Foo:
    def __init__(self, value: int) -> None:
        self.value = value  # never gets run


@dataclass
class Bar(Foo):
    value2: int


Bar(1).value  # runtime error but no mypy error
```
[playground](https://mypy-play.net/?mypy=latest&python=3.10&flags=show-error-context%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cno-implicit-optional%2Cno-implicit-reexport%2Clocal-partial-types%2Cstrict-equality%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=e088e4a40f51c53b1254f04f463e3ced)

not sure if this is supposed to be allowed but either way this fails at runtime, so mypy should warn about it:

> AttributeError: 'Bar' object has no attribute 'value'. Did you mean: 'value2'?

**Comment by DetachHead** [comment]
related: #11809

probably the same issue? but i'm not sure how mypy handles dataclasses
